packages/date-time/src/components/Calendar/CalendarMonth/CalendarMonth.base.tsx

```diff
@@ -59,11 +75,32 @@ export class CalendarMonthBase extends BaseComponent<ICalendarMonthProps, {}> {
     const classNames = getClassNames(styles, {
       theme: theme!,
       className: className,
-      hasHeaderClickCallback: !!this.props.onHeaderSelect,
+      hasHeaderClickCallback: !!onHeaderSelect,
       highlightCurrentMonth: highlightCurrentMonth,
       highlightSelectedMonth: highlightSelectedMonth
     });
 
+    if (this.state.isYearPickerVisible) {
+      return (
+        <CalendarYear
+          minYear={minDate ? minDate.getFullYear() : undefined}
+          maxYear={maxDate ? maxDate.getFullYear() : undefined}
+          onSelectYear={this._onSelectYear}
+          navigationIcons={navigationIcons}
+          onHeaderSelect={this._onYearPickerHeaderSelect}
+          selectedYear={selectedDate ? selectedDate.getFullYear() : navigatedDate ? navigatedDate.getFullYear() : undefined}
+          onRenderYear={this._onRenderYear}
+          strings={{
+            rangeAriaLabel: this._yearRangeToString
+          }}
+          componentRef={this._calendarYearRef}
+          styles={styles}
+          highlightCurrentYear={highlightCurrentMonth}
+          highlightSelectedYear={highlightSelectedMonth}
+        />
+      );
+    }
+
```

Comment:
The way we're jamming CalendarYear inside CalendarMonth bothers me.  The way we have a conditional to return CalendarYear and if not, continue on to return the guts of CalendarMonth has this asymmetry that doesn't feel right.  

CalendarYear and CalendarMonth feel like they should be peers and here CalendarMonth is responsible for CalendarYear.  It certainly doesn't seem intuitive to me